qualtran/bloqs/arithmetic/multiplication.py

```diff
@@ -253,6 +260,7 @@ class ScaleIntByReal(Bloq):
     Args:
         r_bitsize: Number of bits used to represent the real number.
         i_bitsize: Number of bits used to represent the integer.
+        out_bitsizes: The output bitsize for the scaled fixed point real number.
```

Comment:
In the linked reference the output size is assumed to be same as the `r_bitsize`, which is chosen based on the tolerable error `eps`.  

I guess this change in the signature is required to support the usage of this bloq for inverse sqrt where we assume the output bitsize can be specified as an independent parameter?

Reply:
Hmm I might be wrong here but this is more that if we have a real number in [0, 1) (QFxp(n, n), say) and scale it by a a b-bit integer then this would shift the fractional part by at most b so we would need a QFxp(n, n-b) register right? The trotter bloqs also assume the output register can be of a larger size but maybe this casting should be handled separately?

Reply:
> if we have a real number in [0, 1) (QFxp(n, n), say) and scale it by a a b-bit integer then this would shift the fractional part by at most b so we would need a QFxp(n, n-b) register right

Yes, that's correct. And the reference assumes that the output register is always of the size `QFxp(n, n-b)`

> The trotter bloqs also assume the output register can be of a larger size but maybe this casting should be handled separately

I think it's fine to add support for it here - it'd be great if we can update the list of references to include the section in the trotter paper that assumes a different output register size. Do they have corresponding analysis for error propagation? Or is that left as an exercise for the reader?

Reply:
It's a very back of the envelope calculation.